setup.py

```diff
@@ -456,7 +456,36 @@ def build_dlib():
         """use cmake to build and install the extension
         """
         if cmake_path is None:
-            raise DistutilsSetupError("Cannot find cmake in the path. Please specify its path with --cmake parameter.")
+            cmake_install_url = "https://cmake.org/install/"
+            message = ("You can install cmake using the instructions at " +
+                       cmake_install_url)
+            if sys.platform == "darwin":
+                pkgmanagers = ('brew', 'port')
+                for manager in pkgmanagers:
+                    if find_executable(manager) is not None:
+                        message = ("You can install cmake on OSX using {0}"
+                                   " with the command `{0} install cmake`."
+                                   .format(manager))
+                        break
+            else:
+                distname, _, __ = platform.linux_distribution()
```

Comment:
I tried doing that, but apparently on my Ubuntu 14.04:
```
In [10]: platform.linux_distribution()
Out[10]: ('debian', 'jessie/sid', '')
```
There is a bug at http://bugs.python.org/issue9514

If I use that variable directly on my system it'd show "debian" instead of "ubuntu" which may confuse users

I've seen other SO questions like [this](http://stackoverflow.com/questions/30109770/python2-7-pip-install-pymssql-fails-on-centos-6-3) Where centos is wrongly detected as RHEL.

So, I thought it would be a better idea to do this, as that value is not very reliable